src/OpenSpaceToolkit/Astrodynamics/Trajectory/State.cpp

```diff
@@ -499,6 +499,31 @@ State State::inFrame(const Shared<const Frame>& aFrameSPtr) const
     };
 }
 
+bool State::isNear(const State& aState, const std::unordered_map<CoordinateSubset, Real>& aToleranceMap) const
+{
+    if (aToleranceMap.size() != coordinatesBrokerSPtr_->accessSubsets()->getSize())
+    {
+        throw ostk::core::error::runtime::Wrong("Tolerance Map");
+    }
+
+    const State deltaState = aState - *this;
+
+    bool isNear = true;
+    for (const auto& [subset, tolerance] : aToleranceMap) {
+        if (!coordinatesBrokerSPtr_->hasSubset(subset)) {
+            throw ostk::core::error::runtime::Wrong("Tolerance Map Coordinate Subset");
+        }
+
+        const VectorXd deltaCoordinates = deltaState.extractCoordinate(subset);
+
+        if (deltaCoordinates.norm() > tolerance) {
```

Comment:
This makes an inherent assumption that you want to take the norm of each subset. What if you have a composite coordinate subset composed of different units or elements?

Reply:
Might be worth having two different methods, one which does the norm of each subset, and another which compares each element.

Reply:
yeah that's not a bad idea. Maybe we can have a flag that determine which mode this works in, norm or per coordinate